### ISSUE TITLE
TypeError: RSA_PKCS1_PADDING is no longer supported for private decryption

### ISSUE THREAD

**Issue by Metro420yt** [PrismarineJS/node-minecraft-protocol#1293]
[x] The [FAQ](https://github.com/PrismarineJS/node-minecraft-protocol/blob/master/docs/FAQ.md) doesn't contain a resolution to my issue 

## Versions
 - minecraft-protocol: 1.47.0
 - server: vanilla 1.12.2
 - node: 21.7.1

## Detailed description of a problem
when trying to connect, on [login.js line 120](https://github.com/PrismarineJS/node-minecraft-protocol/blob/ccab9fb39681f3ebe0d264e2a3f833aa3c5a1ac7/src/server/login.js#L120) the padding "crypto.constants.RSA_PKCS1_PADDING" throws an error, due to [CVE-2023-46809](https://nodejs.org/en/blog/vulnerability/february-2024-security-releases#nodejs-is-vulnerable-to-the-marvin-attack-timing-variant-of-the-bleichenbacher-attack-against-pkcs1-v15-padding-cve-2023-46809---medium)
```
TypeError: RSA_PKCS1_PADDING is no longer supported for private decryption, this can be reverted with --security-revert=CVE-2023-46809
    at Client.onEncryptionKeyResponse ([...]\minecraft-protocol\src\server\login.js:120:39)
    ...
```

## Current code
```js
createServer({
    version: '1.12.2',
    port: 25565,
    beforePing: (response) => {
        if (properties.enableStatus === false) return;
        response.favicon = favicon;
    },
})
```

## Expected behavior
wasnt sure if there was a way to fix this, or if its just a part of the encryption method

edit: the error is silent, and the server just responds "Disconnected"
and i should clarify, im not sure if its possible to use another encryption method or not 

## Additional context
originally was using node v20.11.1, but updated at some point to see if it would help

~~passing "--security-revert=CVE-2023-46809" gives another error~~
> *was an issue caused by doing `npm audit --force`*

**Comment by extremeheat** [comment]
Seems like a bug in nodejs.

Caused by https://github.com/nodejs/node/commit/54cd268059626800dbe1e02a88b28d9538cf5587

You can open an issue in https://github.com/nodejs/node for it

**Comment by extremeheat** [comment]
Being discussed in https://github.com/nodejs/node/issues/52017

**Comment by extremeheat** [comment]
Proposed programmatic way to re-enable is being discussed in https://github.com/nodejs/node/pull/52090, pure js workaround snippet to fix mentioned in https://github.com/nodejs/node/issues/52017#issuecomment-1987369610

**Comment by IceTank** [comment]
So, it looks like the node team decided to remove the padding feature? Is there a programmatic fix for it that could be added to nmp? I don't want to parse node arguments to all my applications.

**Comment by extremeheat** [comment]
Read the linked issue/PR, I left a comment on https://github.com/nodejs/node/issues/52017#issuecomment-2080906420.

Ideally the standard lib of a language doesn't do breaking updates to remove functionality in minor versions, but that's what seems to be the case at the moment (for security reasons). There is a PR to add a code based toggle to re-enable the funcionality, but it's not for sure that will be merged. This may require using the nodeRSA workaround snippet posted in nodejs/node#52017.

**Comment by IceTank** [comment]
> Read the linked issue/PR, I left a comment on [nodejs/node#52017 (comment)](https://github.com/nodejs/node/issues/52017#issuecomment-2080906420).
> 
> Ideally the standard lib of a language doesn't do breaking updates to remove functionality in minor versions, but that's what seems to be the case at the moment (for security reasons). There is a PR to add a code based toggle to re-enable the funcionality, but it's not for sure that will be merged. This may require using the nodeRSA workaround snippet posted in [nodejs/node#52017](https://github.com/nodejs/node/issues/52017).

I think we should go for the nodeRSA workaround for the time being because Node-Minecraft-Protocol does not work right now with the latest LTS versions of Node.

**Comment by extremeheat** [comment]
Yeah, https://github.com/nodejs/node/issues/52017#issuecomment-1987369610 should have the relevant code to do it
